src/rsa/signing.rs

```diff
@@ -21,6 +21,31 @@ use super::{bigint, bigint::Prime, N, verification};
 use arithmetic::montgomery::{R, RR, RRR};
 use untrusted;
 
+/// Used to construct an RSA key pair from components. Feature: `rsa_signing`.
+///
+/// Components are in big-endian, considered unsigned, and must not contain
+/// leading zeros. To construct a `KeyPair` provide this struct to
+/// `RSAKeyPair::from_key_input()`.
+#[derive(Clone, Debug)]
+pub struct KeyPairInput<'a> {
+    /// Public modulus.
+    pub n: untrusted::Input<'a>,
```

Comment:
It seems like it would be better to instead use a type `der::PositiveInteger` that wraps `untrusted::Input`. Then `der::positive_integer` can return a `PositiveInteger` and each of these fields can be a `der::PositiveInteger`. Then we can document the endianness and lack of leading zeros in the documentation for `der::PositiveInteger` instead of above.

Reply:
A new type would make sense there. Changing the return type of `der::positive_integer()` will have a knock-on effect. Before I go ahead, I just want to confirm that I should change all the associated `untrusted::Input`s to `der::PositiveInteger`s in things like [`ecdsa/verification.rs`](https://github.com/briansmith/ring/blob/b5153f0e388e9b35b7157b3efa93b8091d1876c7/src/ec/suite_b/ecdsa/verification.rs#L27-L30)? (I don't mind chasing the types, I just want to make sure I'm headed in the right direction in case this ends up being a bit invasive.)

Reply:
Yes. In fact, it would be great to have the change of the return type of `der::positive_integer` in a separate PR, that this PR would then be rebased on top of.